### ISSUE TITLE
On Android App Load

### ISSUE THREAD

**Issue by smrinalsingh** [nextcloud/android#13086]
### Cause of error
```java
Exception in thread "main" java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.String.equals(java.lang.Object)' on a null object reference
    at com.owncloud.android.ui.adapter.OCFileListAdapter.limitToPersonalFiles(OCFileListAdapter.java:1074)
    at com.owncloud.android.ui.adapter.OCFileListAdapter.swapDirectory(OCFileListAdapter.java:741)
    at com.owncloud.android.ui.fragment.OCFileListFragment.listDirectory(OCFileListFragment.java:1327)
    at com.owncloud.android.ui.fragment.OCFileListFragment.listDirectory(OCFileListFragment.java:1292)
    at com.owncloud.android.ui.fragment.OCFileListFragment.refreshDirectory(OCFileListFragment.java:1288)
    at com.owncloud.android.ui.activity.FileDisplayActivity.showFiles(FileDisplayActivity.java:2202)
    at com.owncloud.android.ui.activity.DrawerActivity.onNavigationItemClicked(DrawerActivity.java:497)
    at com.owncloud.android.ui.activity.DrawerActivity.lambda$setupDrawerMenu$4(DrawerActivity.java:459)
    at com.owncloud.android.ui.activity.DrawerActivity.$r8$lambda$avHXP34Kh0TDBHMWmR2kRM7pnyQ(Unknown Source:0)
    at com.owncloud.android.ui.activity.DrawerActivity$$ExternalSyntheticLambda3.run(D8$$SyntheticClass:0)
    at android.os.Handler.handleCallback(Handler.java:958)
    at android.os.Handler.dispatchMessage(Handler.java:99)
    at android.os.Looper.loopOnce(Looper.java:257)
    at android.os.Looper.loop(Looper.java:368)
    at android.app.ActivityThread.main(ActivityThread.java:8826)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:572)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1049)

```

### App information
* ID: `com.nextcloud.client`
* Version: `30290090`
* Build flavor: `gplay`

### Device information
* Brand: `OnePlus`
* Device: `OP5155L1`
* Model: `MT2111`
* Id: `UKQ1.230924.001`
* Product: `MT2111_IND`

### Firmware
* SDK: `34`
* Release: `14`
* Incremental: `R.176a8b9_a573-63a5`

**Comment by satvik2131** [comment]
please mention steps to reproduce and screenshots to examine the issue

**Comment by joshtrichards** [comment]
Fixed in #12930
